apps/comps/hooks/useAgents.ts

```diff
@@ -20,38 +26,37 @@ export const useAgents = (params: GetAgentsParams = {}) => {
 };
 
 /**
- * Hook to fetch agents with pagination and filtering
+ * Hook to fetch user's owned agents with pagination
  * @param params Query parameters for agents endpoint
- * @returns Query result with agents data
+ * @returns Query result with user agents data
+
  */
-export const useUserAgents = (params: GetAgentsParams = {}) => {
+export const useUserAgents = (
+  params: GetAgentsParams = {},
+): UseQueryResult<{ userId: string; agents: Agent[] }, Error> => {
   const { isAuthenticated } = useSession();
 
-  return useQuery({
-    queryKey: ["agents", params],
-    queryFn: async (): Promise<AgentsResponse> => {
-      return apiClient.getUserAgents(params);
-    },
-    enabled: isAuthenticated,
-    placeholderData: (prev) => prev,
-  });
+  return useQuery(
+    tanstackClient.agent.getAgents.queryOptions({
+      input: params,
+      enabled: isAuthenticated,
+      placeholderData: (prev) => prev,
```

Comment:
I'm not positive what this does, but if it does what I'm guessing (use the provided data while the query is refetching), that would be cool to prevent flashing of the component.

Reply:
yeah not sure, this is just how the AI translated what was already there